sdk/typespec/src/http/response.rs

```diff
@@ -50,7 +50,7 @@ impl RawResponse {
 
     /// Deserialize the JSON stream into type `T`.
     #[cfg(feature = "json")]
-    pub async fn json<T>(self) -> crate::Result<T>
+    pub fn json<T>(self) -> crate::Result<T>
```

Comment:
Change `json<T>` and `xml<T>` to borrow self rather than consuming self, since they're fully buffered?